### PR TITLE
Add GetDynamicMemberNames() method to DynamicObservableObjectMetaObject

### PR DESCRIPTION
**Pull request by michu**
Hi,

As title says I added GetDynamicMemberNames() to DynamicObservableObjectMetaObject. So now we can get dynamic member names for DynamicObservableObject.

### CODE REVIEW

File: src/Catel.Extensions.DynamicObjects/Catel.Extensions.DynamicObjects.Shared/Data/DynamicObservableObjectMetaObject.cs [hunk 1]
```diff
@@ -28,6 +31,10 @@ static DynamicObservableObjectMetaObject()
 
             _getValueMethodInfo = typeof(DynamicObservableObject).GetMethodEx("GetValue", bindingFlags).MakeGenericMethod(new[] { typeof(object) });
             _setValueMethodInfo = typeof(DynamicObservableObject).GetMethodEx("SetValue", bindingFlags);
+
+            bindingFlags = BindingFlags.Instance | BindingFlags.NonPublic;
```
Comment (GeertvanHorrik) [inline src/Catel.Extensions.DynamicObjects/Catel.Extensions.DynamicObjects.Shared/Data/DynamicObservableObjectMetaObject.cs]: Not sure if private reflection is allowed on limited platforms. Can't we expose this in a different way?

**Comment by michu** [comment]
Yes, the option is to create internal protected method in DynamicObservableObject which will return member names, then dynamically bind to this method. How about this ?

**Comment by GeertvanHorrik** [comment]
Much better, then it will work on .net core as well

**Comment by michu** [comment]
I hope this is it.

File: src/Catel.Extensions.DynamicObjects/Catel.Extensions.DynamicObjects.Shared/Data/DynamicObservableObjectMetaObject.cs [hunk 2]
```diff
@@ -42,6 +43,15 @@ public DynamicObservableObjectMetaObject(Expression parameter, DynamicObservable
         }
 
         /// <summary>
+        /// Returns the enumeration of all dynamic member names.
+        /// </summary>
+        /// <returns>The list of dynamic member names.</returns>
+        public override IEnumerable<string> GetDynamicMemberNames()
+        {
+            return ((DynamicObservableObject)Value).GetPropertyNames();
```
Comment (GeertvanHorrik) [inline src/Catel.Extensions.DynamicObjects/Catel.Extensions.DynamicObjects.Shared/Data/DynamicObservableObjectMetaObject.cs]: No need to safe-cast here? I am not sure, I don't have the full code in my head.

Comment (michu) [inline src/Catel.Extensions.DynamicObjects/Catel.Extensions.DynamicObjects.Shared/Data/DynamicObservableObjectMetaObject.cs]: I'm not sure in 100%, but rather no. Value is passed as parameter before in DynamicObservableObject.GetMetaObject() { return new DynamicObservableObjectMetaObject(parameter, this); } So observable object is value parameter in DynamicObservableObjectMetaObject ctor and is forwarded down to base class. Not sure what happens there, but I think they will not nullify this value.

Comment (GeertvanHorrik) [inline src/Catel.Extensions.DynamicObjects/Catel.Extensions.DynamicObjects.Shared/Data/DynamicObservableObjectMetaObject.cs]: Excellent, thanks.

**Pull request merged**